File: BOJ/DP/2747.py
# 2747 피보나치 수
# https://www.acmicpc.net/problem/2747
import sys
input = sys.stdin.readline

def fibonacci(n):
    f = []
    for i in range(0,n+1):
        if i < 2:
            f.append(1)
        else:
            f.append(f[i-1]+f[i-2])
    return f[n-1]

print(fibonacci(int(input())))

File: BOJ/DP/2747.cpp
// 2747 피보나치 수
// https://www.acmicpc.net/problem/2747
#include <iostream>

using namespace std;

int main(void) {
    int n;
    int dp[46] = {0, };

    cin >> n;

    dp[0] = 0;
    dp[1] = 1;

    for (int i = 2; i <= n; ++i) {
        dp[i] = dp[i-1] + dp[i-2];
    }

    cout << dp[n];

    return 0;
}